azext_iot/operations/hub.py

```diff
@@ -961,42 +993,110 @@ def iot_device_module_key_regenerate(
+            ).response.json()
+        except CloudError as e:
+            handle_service_exception(e)
+        if module["authentication"]["type"] != DeviceAuthApiType.sas.value:
+            raise ClientRequestError("Module authentication should be of type sas")
 
-    pk = module["authentication"]["symmetricKey"]["primaryKey"]
-    sk = module["authentication"]["symmetricKey"]["secondaryKey"]
+        pk = module["authentication"]["symmetricKey"]["primaryKey"]
+        sk = module["authentication"]["symmetricKey"]["secondaryKey"]
 
-    if renew_key_type == RenewKeyType.primary.value:
-        pk = generate_key()
-    if renew_key_type == RenewKeyType.secondary.value:
-        sk = generate_key()
-    if renew_key_type == RenewKeyType.swap.value:
         temp = pk
-        pk = sk
-        sk = temp
+        module["authentication"]["symmetricKey"]["primaryKey"] = sk
+        module["authentication"]["symmetricKey"]["secondaryKey"] = temp
+        try:
+            return service_sdk.modules.create_or_update_identity(
+                id=device_id,
+                mid=module_ids[0],
+                module=module,
+                custom_headers={
+                    "If-Match": '"{}"'.format(etag if etag else "*")
+                },
+            )
+        except CloudError as e:
+            handle_service_exception(e)
 
-    module["authentication"]["symmetricKey"]["primaryKey"] = pk
-    module["authentication"]["symmetricKey"]["secondaryKey"] = sk
+    if renew_key_type in [RenewKeyType.primary.value, RenewKeyType.secondary.value]:
+        renew_key_type += "Key"
 
-    try:
-        headers = {}
-        headers["If-Match"] = '"{}"'.format(etag if etag else "*")
-        return service_sdk.modules.create_or_update_identity(
-            id=device_id,
-            mid=module_id,
-            module=module,
-            custom_headers=headers,
-        )
-    except CloudError as e:
-        handle_service_exception(e)
+    modules = _iot_key_regenerate_process_modules(
+        target, device_id, module_ids
+    )
+    return _iot_key_regenerate_batch(
+        service_sdk=service_sdk,
+        renew_key_type=renew_key_type,
+        items=modules,
+        device_id=device_id
+    )
+
+
+def _iot_key_regenerate_process_modules(
+    target,
+    device_id,
+    module_ids,
+):
+    if module_ids[0] == "*":
+        modules = _iot_device_module_list(target=target, device_id=device_id, top=None)
+        module_ids = []
+        for module in modules:
+            # not going to question why the call the Module object - just making things
+            # easier for unit tests
+            if not isinstance(module, dict):
+                module = module.serialize()
+            if module["authentication"]["type"] == DeviceAuthApiType.sas.value:
+                module_ids.append(module["moduleId"])
+
+    return [{"id": device_id, "moduleId": module_ids[i]} for i in range(len(module_ids))]
```

Comment:
could this be simplified?
```suggestion
    return [{"id": device_id, "moduleId": module_id} for module_id in module_ids]
```

Reply:
idk why I did range